privacy_evaluator/output/user_output.py

```diff
@@ -1,103 +1,43 @@
+import json
 import numpy as np
-import pandas as pd
-from matplotlib import pyplot as plt
-
-from typing import Tuple
 
 
 class UserOutput:
+    def _to_json(self, filter: np.ndarray = None) -> str:
+        """
+        output function for JSON
+        :param filter: if needed this filters the output for the given keys
+        """
+        ret = {}
+        if filter is not None:
+            for key in filter:
+                ret[key] = self._convertToListIfNeeded(self.__dict__.get(key))
+        else:
+            for key, value in self.__dict__.items():
+                ret[key] = self._convertToListIfNeeded(value)
+        return json.dumps(ret)
 
-    """User Output Class"""
-
-    def __init__(
-        self,
-        attack_data_y: np.ndarray,
-        privacy_risk: np.ndarray,
-    ):
+    def _to_dict(self, filter: np.ndarray = None) -> dict:
         """
-        Initilaizes the Class with values
-        :param attack_data_y: An Array of the labels of the attck data
-        :param privacy_risk: the Privacy risk corresponding to the attack data
-        :param all_labels: All labels that are in the training set
+        output function for dicts
+        :param filter: if needed this filters the output for the given keys
         """
-        self.attack_data_y = attack_data_y
-        self.privacy_risk = privacy_risk
+        if filter is not None:
+            ret = {}
+            for key in filter:
+                ret[key] = self.__dict__.get(key)
+        return self.__dict__
```

Comment:
The filtering is ignored here, because you never return `ret`.

Reply:
fixed